internal/app/shortener/strategy/file.go

```diff
@@ -0,0 +1,162 @@
+package strategy
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"path"
+
+	"github.com/Kenny201/go-yandex-shortener.git/internal/domain/shortener/entity"
+	"github.com/Kenny201/go-yandex-shortener.git/internal/domain/shortener/valueobject"
+	"github.com/Kenny201/go-yandex-shortener.git/internal/infra/storage"
+)
+
+var (
+	ErrOpenFile   = errors.New("failed open or create file")
+	ErrDecodeFile = errors.New("failed decode file")
+	ErrEncodeFile = errors.New("failed encode file")
+	ErrCreateDir  = errors.New("failed create or open directory")
+)
+
+type File struct {
+	baseURL    string
+	filePath   string
+	repository Repository
+}
+
+func NewFile(baseURL, filePath string) Strategy {
+	file := &File{}
+	file.filePath = filePath
+	file.repository = storage.NewMemoryShortenerRepository()
+	err := file.ReadAll()
+
+	if err != nil {
+		panic(err)
+	}
+
+	file.baseURL = baseURL
+
+	return file
+}
+
+func (file *File) Get(shortKey string) (*entity.URL, error) {
+	return file.repository.Get(shortKey)
+}
+
+func (file *File) GetAll() map[string]*entity.URL {
+	return file.repository.GetAll()
+}
+
+func (file *File) Put(originalURL string) (string, error) {
+	baseURL, err := valueobject.NewBaseURL(file.baseURL)
+
+	if err != nil {
+		return "", err
+	}
+
+	f, err := os.OpenFile(file.filePath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0666)
+
+	if err != nil {
+		return "", fmt.Errorf("%w: %v", ErrOpenFile, err.Error())
+	}
+
+	encoder := json.NewEncoder(f)
+
+	defer file.close(f)
```

Comment:
Стоит defer вызывать сразу после обработки ошибок.

```suggestion
	defer file.close(f)
	
	encoder := json.NewEncoder(f)
```